src/server/bg_services/lifecycle.js

```diff
@@ -68,13 +68,13 @@ async function handle_bucket_rule(system, rule, j, bucket) {
         })
     });
 
-    //dbg.log0("LIFECYCLE PROCESSING res =", res);
-
     if (res.deleted_objects) {
 
         const writes = [];
 
-        for (const deleted_obj of res.deleted_objects) {
+        for (let i = 0; i < res.deleted_objects.length; ++i) {
+            if (res.delete_fail[i]) continue;
```

Comment:
why do we need this delete_fail? and not checking `if (deleted_obj.err_code !== undefined) continue;` here?

Reply:
if possible to do what I suggested, we want NC lifecycle and lifecycle to share this code, see [comment](https://github.com/noobaa/noobaa-core/pull/8883#discussion_r2006074393)

Reply:
I don't think that returning 2 arrays is a good practice, one with all object infos (even if they are not deleted) and the other with booleans if the delete succeeded or failed - IMO it's very confusing
We need to fix the return value of the RPC request which is incorrect - (was added here the return value was added here - https://github.com/noobaa/noobaa-core/commit/7bc42c2638de859cd35f53cada7b4a8ec8e3e4a5) 
please check the reply of delete_multiple_objects https://github.com/noobaa/noobaa-core/blob/master/src/api/object_api.js#L686-L701
I expected to see a similar reply and not have a different reply structure